src/main/java/core/basesyntax/DataProcess.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DataProcess {
+    private static final int OPERATION = 0;
+    private static final int FRUIT = 1;
+    private static final int NUMBER = 2;
+    private static final String COMMA = ",";
+    private static final char ADD = '+';
+    private static final char SUBTRACT = '-';
+
+    public HashMap<String, Integer> process(String[] data) {
+        HashMap<String, Integer> database = new HashMap<>();
+        for (String datum : data) {
+            String[] check = datum.split(COMMA);
+            if (check[OPERATION].equals(Operation.BALANCE.getOperation())) {
+                database.put(check[FRUIT], Integer.valueOf(check[NUMBER]));
+            }
+            if (check[OPERATION].equals(Operation.SUPPLY.getOperation())
+                    || check[OPERATION].equals(Operation.RETURN.getOperation())) {
+                mapIterator(database, ADD, check);
+            }
+            if (check[OPERATION].equals(Operation.PURCHASE.getOperation())) {
+                mapIterator(database, SUBTRACT, check);
+            }
```

Comment:
that's not how we implement strategy. We should do it via HashMap, where operation is key, and operationHandler is value, it will be easier to extend later